gnd/src/main/java/com/google/android/gnd/ui/common/BindingAdapters.java

```diff
@@ -118,4 +125,29 @@ public static void setOnFocusChangeListener(
   public static void setOnShowDialogListener(MultipleChoiceFieldLayout view, Runnable listener) {
     view.setOnShowDialogListener(listener);
   }
+
+  @BindingAdapter({"bind:storageManager", "bind:response"})
+  public static void bindPhoto(ImageView view, StorageManager storageManager, Response response) {
```

Comment:
This is a lot of logic to be doing inside a binding adapter, especially the hiding/showing of the parent view. Can we instead define a separate view widget that can be tested and implemented independently? It could have its own ViewModel with LiveData containing the actual image to display. Wdyt?

Reply:
Makes sense. Then we can reuse it at other places as well.

Reply:
I've added a custom view + a view model for it. But this is again having the original problem of image view not getting updated (nor the visibility of it's container). Can you please have a look? I tried debugging it but couldn't find anything useful.

Reply:
Great job finding this issue, I looked over the weekend but didn't find it!

Do you think we can let the binding take a Uri instead of a StorageManager and response to prevent having so much logic in the binder?